common/src/main/java/com/wynntils/features/user/AutoApplyResourcePackFeature.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigManager;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.mc.event.ResourcePackClearEvent;
+import com.wynntils.mc.event.ResourcePackEvent;
+import com.wynntils.mc.event.TitleScreenInitEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.model.WorldStateManager;
+import java.io.File;
+import java.util.Objects;
+import net.minecraft.client.resources.ClientPackSource;
+import net.minecraft.server.packs.repository.PackSource;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class AutoApplyResourcePackFeature extends UserFeature {
+    private final File SERVER_RESOURCE_PACK_DIR = new File(McUtils.mc().gameDirectory, "server-resource-packs");
+
+    @Config(visible = false)
+    private String packHash = "";
+
+    private String appliedHash = "";
```

Comment:
What is the difference between the two?

Reply:
The config is to save the last pack hash Wynn used, the second is to keep track of the currently loaded pack. Needed if Wynn changes pack's mid-game, the game won't be in a weird state.

Reply:
(otherwise I could use a bool)

Reply:
Then what does 
```java

        for (File file : files != null ? files : new File[0]) {
            if (clientPackSource.checkHash(packHash, file)) {
                clientPackSource.setServerPack(file, PackSource.DEFAULT);
                appliedHash = packHash;
            }
        }
```
do?

Reply:
Find which file to load in the server resources directory

Reply:
Isn't `appliedHash` more of the local hash we have on the client?

Reply:
appliedHash is the same as packHash if we have pre-loaded the pack. Otherwise it is empty.